openedx_tagging/core/tagging/models/base.py

```diff
@@ -260,87 +295,196 @@ def copy(self, taxonomy: Taxonomy) -> Taxonomy:
         self._taxonomy_class = taxonomy._taxonomy_class  # pylint: disable=protected-access
         return self
 
-    def get_tags(
+    def get_filtered_tags(
         self,
-        tag_set: models.QuerySet[Tag] | None = None,
-    ) -> list[Tag]:
+        depth: int | None = TAXONOMY_MAX_DEPTH,
+        parent_tag_value: str | None = None,
+        search_term: str | None = None,
+        include_counts: bool = True,
```

Comment:
I think that any time you're editing tags or taxonomies, it's helpful to see which tags are being used and their relative popularity. Especially when selecting a tag from a list, to tag a content item, it helps to see which tags are more commonly used. That can help distinguish between similar tags.

I left it on by default since I think it's useful and it doesn't seem to slow the query down by more than a millisecond but it's true that we're not using it in the current UI so it could make sense to have it off by default.

Reply:
Hmm.. that does makes sense.. it'll be useful for free-text tags, for sure. But I don't think we need it on right now.

Reply:
OK, I made it off by default.